rstudio/rstudio
# Add commands for showing all panes

### Intent

Fixes #5775

Keyboard-only users need a way to activate all the panes in the UI. Many of the ephemeral panes shown in the Console panel did not have these.

Also, having a direct way to show all of these (sometimes rare) panes is helpful for accessibility testing (and testing in general).

### Approach

Added commands for these, in a new submenu of the View menu ("Show Other Panes"). Also makes them available in the Command Palette, of course.

Moved the existing "Show Find in Files" command to this submenu, and added the existing command (which wasn't in the main menu) for activating Presentations pane to the menu, but only shown if the Presentation pane exists. Similar behavior already happens for the Build tab.

### QA Notes

This PR only covers showing the panes, ensuring we are correctly giving them keyboard focus is tracked separately in #5652.

Here is the fully populated menu (from a Package project that has a Presentation in it). Confirm that "Show Build" and "Show Presentation" only appear when in a buildable-project and/or you have previewed a presentation. The rest of the menu items will always appear.

<img width="411" alt="screenshot of View menu showing new Show Other Panes submenu" src="https://user-images.githubusercontent.com/10569626/92636848-e59f7f80-f28c-11ea-9cdd-e97b10a59adf.png">

Here's how the Console panel looks with every pane displayed (open source, Pro would add the Launcher tab, too). This isn't new, just harder to do prior to adding these commands. If you need instructions on how to make these appear without using the new menu commands, let me know, I've got notes.

<img width="933" alt="Console panel showing all the possible panes" src="https://user-images.githubusercontent.com/10569626/92637216-678fa880-f28d-11ea-8fd8-eb43d3daab9d.png">

Verify that the commands display and select the correct pane. They should also restore the Console panel if it has been minimized.
